### ISSUE TITLE
Would Prefer Blocking Apply to Frames As Well

### ISSUE THREAD

**Issue by SleepExpert** [log69/yesscript2#3]
The primary way I used YesScript was to block dozens of domains that are almost exclusively loaded in iframes. YesScript2 appears to only block scripts for the top-level frame ("main_frame"). This extension would work much better for me if it also checked the domains of scripts loaded in iframes ("[sub_frame](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/webRequest/ResourceType)").

Even if this (apply to sub_frame) was an option that could only be enabled by manually editing storage.js, it would still suit my needs (worth 5-stars).

Because the developer has been awesome enough to provide the source, I can create my own, altered version of the extension that would work the way I need but I think many people would want this option (maybe even as the default functionality) so I figured I mention it here first.

**Comment by log69** [comment]
Sounds reasonable, I'll take a look at sub_frame and make it if possible.

**Comment by log69** [comment]
I've looked into the documentation and adding "sub_frame" to the webrequest type should really suffice it to work. Better this way really. Done it in git, made it default, the extension should also update to v2.5 soon.

Upgrade it in your browser and I'd gladly take a feedback from you whether it works as expected. I've also made tests with scripts within iframes and seems fine. See the diff here:

https://github.com/log69/yesscript2/commit/3da8c15e860d7bc3fe86d1b4be28ceea3c70d8b1?diff=unified
